plutus-pab/src/Cardano/Node/Mock.hs

```diff
@@ -115,29 +119,32 @@ runChainEffects ::
 runChainEffects serverHandler clientHandler stateVar eff = do
     oldAppState <- liftIO $ takeMVar stateVar
     ((a, events), newState) <- liftIO
-        $ runM
-        $ runStderrLog
-        $ interpret renderLogMessages
-        $ Eff.runState oldAppState
-        $ Eff.runReader serverHandler
-        $ Eff.runReader clientHandler
-        $ Eff.runWriter
-        $ reinterpret (handleLogWriter @MockServerLogMsg @[LogMessage MockServerLogMsg] (unto return))
-        $ interpret (handleZoomedState chainState)
-        $ interpret (handleZoomedState followerState)
-        $ interpret (mapLog ProcessingChainEvent)
-        $ reinterpret CE.handleChain
-        $ interpret (mapLog ProcessingChainEvent)
-        $ reinterpret Chain.handleControlChain
-        $ interpret (mapLog FollowerMsg)
-        $ FE.handleNodeFollower
-        $ subsume
-        $ runGenRandomTx
-        $ do result <- eff
-             void Chain.processBlock
-             pure result
+            $ processBlock eff
+            & runRandomTx
+            & runNodeFollwer
+            & runChain
+            & mergeState
+            & toWriter
+            & runReaders oldAppState
+            & interpret renderLogMessages
+            & runStderrLog
+            & runM
     liftIO $ putMVar stateVar newState
     pure (events, a)
+        where
+            processBlock e = e >>= \r -> Chain.processBlock >> pure r
+
+            runRandomTx = subsume . runGenRandomTx
+
+            runNodeFollwer = interpret (mapLog FollowerMsg) . FE.handleNodeFollower
```

Comment:
I really wanted to add short concise type signatures to these bindings indicating which effects are being peeled off. Something sort of like the following:

```haskell
runRandomTx :: (Member GenRandomTx  effs) => Eff (GenRandomTx ': effs) ~> Eff effs
runRandomTx = _
```

But i guess i would have to use the explicit `NodeServerEffects` type somehow and then..all of this doesn't really work. Anyway, beats me.

Reply:
Yeah that's the problem with defining lists of effects :(

Reply:
With the type application after `interpret` you get minimal noise - just the effect that is handled here.